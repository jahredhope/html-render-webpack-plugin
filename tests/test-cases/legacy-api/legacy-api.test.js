const { createInMemoryFileSystem } = require("../../utils/memory-fs");
const webpack = require("webpack");
const path = require("path");

const config = require("./webpack.config");

describe("Use legacy api", () => {
  it("should render a HTML file", async (done) => {
    const compiler = webpack(config);

    const memoryFs = createInMemoryFileSystem();
    compiler.outputFileSystem = memoryFs;

    compiler.run((error) => {
      expect(error).toBe(null);
      const contents = memoryFs.readFileSync(
        path.join(process.cwd(), "dist", "index.html"),
        "utf8"
      );
      expect(contents).toMatchSnapshot();
      done();
    });
  });
});
