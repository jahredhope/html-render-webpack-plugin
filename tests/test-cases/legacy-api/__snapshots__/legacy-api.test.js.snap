// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use legacy api should render a HTML file 1`] = `
"<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"/\\",
      \\"extra\\": \\"extra-value\\",
      \\"assets\\": {
        \\"main\\": \\"client-main-97b069c18a91450b740e.js\\"
      }
    }
  ]</code>
</body>
</html>"
`;
